src/circuit.rs

```diff
@@ -276,9 +277,14 @@ impl plonk::Circuit<pallas::Base> for Circuit {
                             - v_new.clone()
                             - magnitude * sign,
                     ),
+                    // We already checked that
+                    // * split_flag is boolean (just above), and
+                    // * v_old is a 64 bit integer (in note commitment evaluation).
+                    // So, split_flag + v_old = 0 only when both split_flag and v_old are equal to
+                    // zero (no overflow can occur).
```

Comment:
So, split_flag + v_old = 0 only when (split_flag = 0 and v_old = 0), no overflow can occur.

Reply:
Done